chain/trie/sync.go

```diff
@@ -0,0 +1,188 @@
+package trie
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/nknorg/nkn/v2/chain/db"
+	"github.com/nknorg/nkn/v2/common"
+	"github.com/nknorg/nkn/v2/common/prque"
+)
+
+type SyncPath []byte
+
+// request represents a scheduled or already in-flight state retrieval request.
+type request struct {
+	path []byte         // Merkle path leading to this node for prioritization
+	data []byte         // Data content of the node, cached until all subtrees complete
+	hash common.Uint256 // Hash of the node data content to retrieve
+}
+
+// schedule inserts a new state retrieval request into the fetch queue
+func (s *Sync) schedule(req *request) {
```

Comment:
Better to put method after Sync declaration. Let's follow this convention:

type definition

type constructor

type public method

type private method